packages/gatsby/src/query/index.js

```diff
@@ -249,17 +250,36 @@ const runQueuedQueries = () => {
  *
  * For what constitutes a dirty query, see `calcQueries`
  */
-const startListening = queue => {
+const startListeningToDevelopQueue = queue => {
   // We use a queue to process batches of queries so that they are
   // processed consecutively
+  let graphqlRunner = null
+  const developQueue = queryQueue.createDevelopQueue(() => {
+    if (!graphqlRunner) {
+      graphqlRunner = new GraphQLRunner(store)
+    }
+    return graphqlRunner
+  })
   listenerQueue = new Queue((queryJobs, callback) =>
     queryQueue
-      .processBatch(queue, queryJobs)
+      .processBatch(developQueue, queryJobs)
       .then(() => callback(null))
       .catch(callback)
   )
 
   emitter.on(`API_RUNNING_QUEUE_EMPTY`, runQueuedQueries)
+  ;[
+    `DELETE_CACHE`,
+    `CREATE_NODE`,
+    `DELETE_NODE`,
+    `DELETE_NODES`,
+    `SET_SCHEMA_COMPOSER`,
+    `SET_SCHEMA`,
```

Comment:
Do we need to also handle `ADD_FIELD_TO_NODE` and `ADD_CHILD_NODE_TO_PARENT_NODE`?

Or can we remove these two in a later version of Gatsby?